tests/integration_tests/functional/test_balloon.py

```diff
@@ -22,10 +22,22 @@
 MB_TO_PAGES = 256
 
 
-def get_rss_mem_by_pid(pid):
+def get_stable_rss_mem_by_pid(pid):
     """Get the RSS memory that a guest uses, given the pid of the guest."""
-    output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
-    return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+
+    def get_rss_from_pmap():
+        output = subprocess.check_output("pmap -X {}".format(pid), shell=True)
+        return int(output.decode('utf-8').split('\n')[-2].split()[1], 10)
+
+    rss = get_rss_from_pmap()
+    percentage_delta = 100
+    while percentage_delta > 0.5:
+        prev_rss = rss
+        rss = get_rss_from_pmap()
```

Comment:
There is a chance that both `pmem` calls return the same value (or similar values) and the while loop exits early. I think that the `time.sleep(1)` command should be executed before the second `pmem` call.

Reply:
Oh yes, nice catch, thanks!